yargs.js

```diff
@@ -301,42 +301,58 @@ function Yargs (processArgs, cwd, parentRequire) {
     return self
   }
 
-  self.demand = self.required = self.require = function (keys, max, msg) {
+  self.demandOption = self.demand = self.required = self.require = function (keys, max, msg) {
```

Comment:
> `demandOption` doesn't have all the features the old `demand` did.
@maxrimue yea, you're right; people will be expecting the same behaviour that we had before. let me change this